servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/TaskBasedAsyncPublisherOperator.java

```diff
@@ -102,13 +118,15 @@ protected void handleSubscribe(Subscriber<? super T> subscriber,
         @Nullable
         private Subscription subscription;
 
-        OffloadedSubscriber(final Subscriber<? super T> target, final Executor executor) {
-            this(target, executor, 2);
+        OffloadedSubscriber(final Subscriber<? super T> target,
+                            final BooleanSupplier offload, final Executor executor) {
+            this(target, offload, executor, 2);
         }
 
-        OffloadedSubscriber(final Subscriber<? super T> target, final Executor executor,
+        OffloadedSubscriber(final Subscriber<? super T> target, final BooleanSupplier offload, final Executor executor,
                             final int publisherSignalQueueInitialCapacity) {
             this.target = target;
+            this.offload = offload;
             this.executor = executor;
             // Queue is bounded by request-n
             signals = newUnboundedSpscQueue(publisherSignalQueueInitialCapacity);
```

Comment:
line 143: `public void onNext(final T t)` -> `public void onNext(@Nullable final T t)`

Reply:
also plz remove debug/development logs below (`LOGGER.trace("offloading Publisher onSubscribe on {}", executor);`, ..). operators allow for instrumentation/visibility if required.